core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala

```diff
@@ -97,26 +104,106 @@ class BpmBillingProjectLifecycle(
             RawlsErrorReport(StatusCode.int2StatusCode(errorReport.getStatusCode), errorReport.getMessage)
           )
         case None =>
+          landingZoneId = Some(landingZoneResponse.getLandingZoneId)
           logger.info(
-            s"Creating BPM-backed billing project ${projectName}, initiated creation of landing zone ${landingZoneResponse.getLandingZoneId} with jobId ${landingZoneResponse.getJobReport.getId}"
+            s"Creating BPM-backed billing project ${projectName.value}, initiated creation of landing zone ${landingZoneId.get} with jobId ${landingZoneResponse.getJobReport.getId}"
           )
       }
-      for {
-        _ <- billingRepository.updateLandingZoneId(createProjectRequest.projectName,
-                                                   landingZoneResponse.getLandingZoneId
-        )
+      (for {
+        _ <- billingRepository.updateLandingZoneId(createProjectRequest.projectName, landingZoneId.get)
         _ <- resourceMonitorRecordDao.create(
           UUID.fromString(landingZoneResponse.getJobReport.getId),
           JobType.AzureLandingZoneResult,
-          projectName
+          projectName.value
         )
         _ <- billingRepository.setBillingProfileId(createProjectRequest.projectName, profileModel.getId)
-      } yield CreationStatuses.CreatingLandingZone
+      } yield CreationStatuses.CreatingLandingZone).recoverWith { case t: Throwable =>
+        for {
+          _ <- cleanupLandingZone(landingZoneId, projectName, ctx)
+          _ <- cleanupBillingProfile(profileModelId, projectName, ctx)
+        } yield throw t
+      }
     } catch {
-      case exception: Exception => Future.failed(exception)
+      case t: Throwable =>
+        for {
+          _ <- cleanupLandingZone(landingZoneId, projectName, ctx)
+          _ <- cleanupBillingProfile(profileModelId, projectName, ctx)
+        } yield throw t
     }
   }
 
+  private def cleanupLandingZone(landingZoneId: Option[UUID],
+                                 projectName: RawlsBillingProjectName,
+                                 ctx: RawlsRequestContext
+  ): Future[Unit] =
```

Comment:
This method could have just returned `Unit`, but for symmetry with `cleanupBillingProfile` I made it return `Future[Unit]`.

Reply:
I am of the mind that anything making an external call should really return `Future` (or better yet,`IO`).